### ISSUE TITLE
Please make the 'news' button less obtrusive for registered users

### ISSUE THREAD

**Issue by LinuxOnTheDesktop** [RealRaven2000/QuickFolders#313]
I value your software, I appreciate too that whenever there is 'news', there is no longer a pop-up. Yet, I do not want to see the bright, large news button _either_. For, I have paid for this software and I want merely to use it. I do not want to have to quieten my UI by clicking a button every time that you update your software. Thanks,

**Comment by RealRaven2000** [comment]
Thanks for the suggestion, I have to think this through. While it may seem trivial to just hide the "news" away thing like the last maintenance release (5.13.1) with an issue like #312 which generated garbage emails and I had to fix basically on the day it was reported (1 day after releasing 5.13) can be very very important to some users. While I have no problem offering an option for users with valid licenses to hide all news I also believe it is important to communicate changed behavior somehow. I could make the visual cue a lot less obtrusive (I was thinking of a little "badge" icon that's superimposed on the QuickFolders button) but I do also use the time period around new versions for tell my users that there are cheaper license upgrades / licenses available. 

It can also be relevant if your license is going to run out within a few days and you would miss a opportunity to save some money (even if it's just 2 bucks), so I will have to make do some modifications that are a lot more nuanced than just "hide news".

Whatever the new feature will look like it will likely go into the newish "Updates" panel on the Advanced settings tab:

![image](https://user-images.githubusercontent.com/6314674/193258978-47bd73bc-40a2-4818-acac-fc036d795d6c.png)

It might be an opt in function like [x] indicate update / news less prominently  (not sure how to word this yet). I believe there should still be a minimum visibility of new behavior / functions, e.g. like that:

![image](https://user-images.githubusercontent.com/6314674/193260422-4dc19584-4e3b-41a1-8c8c-f3cd286fff0a.png)

Disclaimer: On my own system I have configured the QuickFolders label to be hidden in day to day operation but obviously I am making it visible whenever I install a new test version (and that may be 100 times before I release a final version), so I definitely understand how it sucks to have something taking screen space that you do not want to be there. Bear in mind I have gradually removed anything that could be perceived as distraction (version tab, automatic news popup) in order to get out of my users way and I am constantly thinking about how I could make QF more useful, fix highly fundamental problems and keep the financing of the work running in the background as smoothly as possible.

**Comment by fredstluka** [comment]
I disagree with this request.  As a paid registered user, I prefer to be notified when QF updates itself, and given a chance to see the list of changes.

Please do NOT make this change, except perhaps as a new option for those who prefer fewer notifications.

Thanks!
--Fred

**Comment by crablouis** [comment]
> I disagree with this request. As a paid registered user, I prefer to be notified when QF updates itself, and given a chance to see the list of changes.
<SNIP>

Same. 

I also prefer a clear up front indicator of updates and changes with a super easy link to the change details.

Silent updating could, and should be an option, as Axel suggests.

**Comment by RealRaven2000** [comment]
> I disagree with this request. As a paid registered user, I prefer to be notified when QF updates itself, and given a chance to see the list of changes.
> 
> Please do NOT make this change, except perhaps as a new option for those who prefer fewer notifications.

Well that was my whole point as I described it as "opt-in". For the alert user a little red `[News]` badge may be just sufficient, especially if they like to key their UI clean.  I think it will be nice as well to mix with the "Renew License" style (which overrides the news badge at the moment). Also, there is another problem with the animation on the Renew License (pulsing shadow) it actually slows down Thunderbird on some OS  - see issue #310. So I might address both items together as part of the Updates option group.

The [Add-on compatibility check](https://addons.thunderbird.net/thunderbird/addon/addon-compatibility-check/) extension has a similar UI where it super-imposes the number of incompatible icons in a red badge:

![image](https://user-images.githubusercontent.com/6314674/193451337-3c71441e-6433-4e66-81ad-0410ab3d3dc6.png)

I think this is quite slick, and if it is opt-int we can just use a small font for the news badge. I might even include a [sale] badge for expired (or soon to expire) licenses, provided there is a sale active.

**Comment by fredstluka** [comment]
Axel, 

Opt-in, small badge, sale badge, warning of coming expiration, etc. -- All sounds good to me.  Keep up the good work!

--Fred

**Comment by RealRaven2000** [comment]
**Current Behavior**

If the QuickFolders label is displayed, there is no styling (just default text on default background color)
![image](https://user-images.githubusercontent.com/6314674/193553586-45743b6a-d126-4e82-8503-0a28beaabb5a.png)

Once an update happens - the text & style changes - it is also forced visible even if the label was configured "away". I change the text to something much longer (sufficiently annoying to make the user push it at one stage, and I think this may come across as disruptive for those who prefer "silent updates") 
![image](https://user-images.githubusercontent.com/6314674/193553792-bb96991b-da04-4cc3-a4bc-4b9b2d25e7f5.png)


Once the license expires:
![image](https://user-images.githubusercontent.com/6314674/193549293-bf8703fc-00f6-43a1-bc66-4a3f9410ef28.png)

When a new version lands with an expired license, I change the text as above - but I do not change the red color (to indicate the license is expired)

![image](https://user-images.githubusercontent.com/6314674/193551101-c0c2c38f-5050-46d2-b9fd-33818649a912.png)

**New Behavior**

With the new behavior, this could be something like this instead :
![image](https://user-images.githubusercontent.com/6314674/193550298-b57a6f7c-c9aa-4416-a5a6-2e2e8fa3f63b.png)

The red color of the button + limited space works against me a little here. Maybe there is space to add a white outline to the badge
![image](https://user-images.githubusercontent.com/6314674/193552600-1eb43091-ca86-4fad-8b42-f3e3d27d72f9.png)

...even as opt in, this is already a lot better UI experience.

If there is no expired License, how should I display the badge?
![image](https://user-images.githubusercontent.com/6314674/193554688-a86fed88-bbb7-4c03-9c08-11a7b74fbd1e.png)

Alternative - use the "blue button" style:
![image](https://user-images.githubusercontent.com/6314674/193555806-edce5288-b4b0-48c5-880e-ab2a3d1d05e1.png)

I like that one. This variant seems to provide a nicer contrast between the blue and hot pink and attracts the eye without being totally distracting. Of course not flashing / throbbing will be included in this solution. Thoughts?

By the way, if the user has no license at all I would probably use the old style, or should I allow anybody to use this feature? On the one hand I like a slicker UI if it doesn't lead to be ignored, on the other hand I know that a lot of more casual users can ignore all other tabs than the main one, so an extra 20px of badge will not be noticed at all...

**Comment by RealRaven2000** [comment]
One more addendum I could just change the News label to SALE if there is money to be saved:

![image](https://user-images.githubusercontent.com/6314674/193556994-671f4e90-e2ff-43d1-8dc0-e48bcc318b0a.png)

But the way that it works at the moment, you only need to click on the button once (to view the change list) and then the badge is gone forever. So the question is, should the SALE sign stay up until the offer is expired? That would be good for sustainability on the one hand, but on the other hand it would feel like a broken feature (click once only to acknowledge the software has changed).

**Comment by RealRaven2000** [comment]
Dd-ons developer meeting yesterday there was an outline on how to achieve the badge look

- create new custom element based on code for `MozToolbarbutton`
 
`CustomElements.define("toolbarbutton-badged", MozToolbarButton)`

https://searchfox.org/comm-central/source/mozilla/toolkit/content/widgets/toolbarbutton.js#33

**Comment by xovaox** [comment]
Definitely it has to be done, the current behaviour displaying a notification each time QuickFolder is just very annoying; I like silent things, with no intervention from my side, even the smaller click on an icon; I have really other things with more interest to do.

**Comment by RealRaven2000** [comment]
Just a heads up - I haven't forgotten about this one, and still want it, but at the moment the compatibility work for Thunderbird 115 has taken precedence. I have been testing with Thunderbird 113 betas and there is a big amount of breakage that needs to be dealt with. Also a lot of the user interface regarding buttons has to be rewritten, so I am not going to make a big investment into this (news badge) feature on Thunderbird 102. It will have to wait until Thunderbird 115.

See also:  issue #351 - QuickFolders
quickFilters: [issue 181](https://github.com/RealRaven2000/quickFilters/issues/181)
SmartTemplates: [issue 213](https://github.com/RealRaven2000/SmartTemplates/issues/213)
FiltaQuilla: [issue 223](https://github.com/RealRaven2000/FiltaQuilla/issues/223)

**Comment by LinuxOnTheDesktop** [comment]
Thanks, RealRaven. It might be useful for others to follow [this link to a Mozilla blogpost about Thunderbird 113](https://blog.thunderbird.net/2023/02/thunderbird-115-supernova-preview-the-new-folder-pane/) (but this bugtracker is not the place to discuss, or anticipate, the virtues or vices of the forthcoming new Thunderbird interface).

**Comment by LinuxOnTheDesktop** [comment]
I notice the following behaviour in QF 6.11.1 (and earlier?). The QF news appears when one updates QF and, after one has closed the news windows, there is still a big, highlighted 'what's new' button (or some such wording along those lines) in the tab bar. That big highlighted button goes away only when, by clicking on it, one views the news, _again_.

It would be better for the user where the news-advertising-button not to appear after one has, by any route, read the news. Thanks.

**Comment by RealRaven2000** [comment]
Implemented in 6.13 - Published 17/09/2025

**Comment by RealRaven2000** [comment]
For reference, please see #607
